astropy/io/fits/column.py

```diff
@@ -2088,22 +2090,14 @@ def _scalar_to_format(value):
         except ValueError:
             continue
 
-    if isinstance(value, int) and value in (0, 1):
-        # Could be a boolean
-        return 'L'
-    elif isinstance(value, int):
-        for char in ('B', 'I', 'J', 'K'):
-            type_ = np.dtype(FITS2NUMPY[char]).type
-            if type_(value) == value:
-                return char
-    elif isinstance(value, float):
-        # For now just assume double precision
-        return 'D'
-    elif isinstance(value, complex):
-        return 'M'
-    else:
-        return 'A' + str(len(value))
+    numpy_dtype_str = np.min_scalar_type(value).str
+    numpy_dtype_str = numpy_dtype_str[1:] #Strip endianness
```

Comment:
This is a terrible nitpick but there should be two spaces before the `#` and one space after.